narwhals/_pandas_like/group_by.py

```diff
@@ -160,15 +160,16 @@ def func(df: Any) -> Any:
             out_group, index=out_names, name="", implementation=implementation
         )
 
-    if implementation == "pandas":
-        pd = get_pandas()
+    extra_args: dict[str, Any] = {}
+    if implementation is Implementation.PANDAS:  # pragma: no cover
+        backend = get_backend(implementation)
+        backend_version = parse_version(backend.__version__)
+        minimum_version = parse_version("2.2.0")
+
+        if backend_version >= minimum_version:
+            extra_args["include_groups"] = False
```

Comment:
(nitpicking, sorry) suggesting variable names and usage that I'd find more immediate/readable

```suggestion
    apply_kwargs: dict[str, Any] = {}
    if implementation is Implementation.PANDAS:  # pragma: no cover
        backend = get_backend(implementation)

        if parse_version(backend.__version__) >= parse_version("2.2.0"):
            apply_kwargs["include_groups"] = False
```